### DISCUSSION TITLE
How do I build using clang on windows?

### DISCUSSION THREAD

**Discussion by rupakhetibinit** [friendlyanon/cmake-init#116] [Q&A] [answered]
By default it seems to use MSVC. I don't see any option to change it.

**Comment by friendlyanon** [comment]
Using LLVM Clang requires nothing more than setting it as the compiler.

**Reply by rupakhetibinit** [reply]
Yes. But that's precisely the problem I'm facing. I'm unsure how do I set it as the default. On the information I've looked over the internet, it seems like setting env vars should work but It just doesn't seem to work. 
$env:CC="C:\Users\BINIT\scoop\apps\llvm\current\bin\clang.exe" 
$env:CXX="C:\Users\BINIT\scoop\apps\llvm\current\bin\clang++.exe" 
This is how I'm setting these in env vars in powershell

**Reply by friendlyanon** [reply, marked as answer]
The Visual Studio generator doesn't care about those env variables. Use Ninja and the [CMAKE_\<LANG\>_COMPILER](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html) cache variable. Make a new user preset for convenience.
